src/server/worker/pipeline/transform/status.go

```diff
@@ -98,3 +101,20 @@ func (s *Status) Cancel(jobID string, datumFilter []string) bool {
 	}
 	return false
 }
+
+func (s *Status) NextDatum(ctx context.Context, err error) ([]string, error) {
+	if s.nextChan == nil {
```

Comment:
It seems like this could race with the assignment in the other file.

Reply:
Refer to the comment above.